Src/FluentAssertions/Common/Clock.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace FluentAssertions.Common
+{
+    /// <summary>
+    /// Default implementation for <see cref="IClock"/> for production use.
+    /// </summary>
+    internal class Clock : IClock
```

Comment:
Why is it internal?
How to use it in extensions?

Reply:
It was already `internal`. I'm not sure the API is stable enough to be exposed as an official extension point. People can always provide their own implementation. What do you think @jnyrup ?

Reply:
If you think about the stability of the API, the interfaces should be internal as well.

Reply:
Agreed, but they are part of the constructor dependencies of classes that are already public. In short, we can't.